Sources/OAuthenticator/Authenticator.swift

```diff
@@ -20,7 +20,8 @@ public enum AuthenticatorError: Error {
 /// Manage state required to executed authenticated URLRequests.
 public final class Authenticator {
 	public typealias UserAuthenticator = (URL, String) async throws -> URL
-
+    public typealias AuthenticationResult = (Login?, AuthenticatorError?) -> Void
```

Comment:
Two questions for this type.

First, what do you think about making the parameter `Result<Login, AuthenticatorError>`? This would help to ease the optional juggling.

Second, I keep coming back to the name `AuthenticationResult`. What do you think about something like `DidLogin` or `LoginHandler`?

Reply:
1. Not sure why but I never use the `Result` type. However I think this is much better. Will make the change.

2. I feel that there are a lot of `Provider` / `Handler` like types in the package. Do you think `DidLogin` would go against the naming convention? Also `LoginHandler` feels that you *need* to handle the login which is not the case. It's only a reporting mechanism. 
What about `AuthenticationStatusHandler` or something like that?

Reply:
1. Great! It took me a while to warm up to `Result` too. And, I often now prefer throwing, since it fits in much better with concurrency. But in this case, it does seem like a good fit.

2. Yeah, I didn't think too hard, but you are making a good point. That name works for me!